src/contracts/core/DelegationManagerStorage.sol

```diff
@@ -13,37 +12,76 @@ import "../interfaces/ISlasher.sol";
         keccak256("EIP712Domain(string name,uint256 chainId,address verifyingContract)");
 
-    /// @notice The EIP-712 typehash for the delegation struct used by the contract
-    bytes32 public constant DELEGATION_TYPEHASH =
-        keccak256("Delegation(address delegator,address operator,uint256 nonce,uint256 expiry)");
+    /// @notice The EIP-712 typehash for the StakerDelegation struct used by the contract
+    bytes32 public constant STAKER_DELEGATION_TYPEHASH =
+        keccak256("StakerDelegation(address staker,address operator,uint256 nonce,uint256 expiry)");
 
-    /// @notice EIP-712 Domain separator
-    bytes32 public DOMAIN_SEPARATOR;
+    /// @notice The EIP-712 typehash for the DelegationApproval struct used by the contract
+    bytes32 public constant DELEGATION_APPROVAL_TYPEHASH =
+        keccak256("Delegation(address staker,address operator,uint256 nonce,uint256 expiry)");
+
+    /**
+     * @notice Original EIP-712 Domain separator for this contract.
+     * @dev The domain separator may change in the event of a fork that modifies the ChainID.
+     * Use the getter function `domainSeparator` to get the current domain separator for this contract.
+     */
+    bytes32 internal _DOMAIN_SEPARATOR;
 
     /// @notice The StrategyManager contract for EigenLayer
     IStrategyManager public immutable strategyManager;
 
     /// @notice The Slasher contract for EigenLayer
     ISlasher public immutable slasher;
 
-    /// @notice Mapping: operator => strategy => total number of shares in the strategy delegated to the operator
+    /**
+     * @notice returns the total number of shares in `strategy` that are delegated to `operator`.
+     * @notice Mapping: operator => strategy => total number of shares in the strategy delegated to the operator.
+     */
     mapping(address => mapping(IStrategy => uint256)) public operatorShares;
 
-    /// @notice Mapping: operator => delegation terms contract
-    mapping(address => IDelegationTerms) public delegationTerms;
+    /**
+     * @notice Mapping: operator => OperatorDetails struct
+     * @dev This struct is internal with an external getter so we can return an `OperatorDetails memory` object
+     */
+    mapping(address => OperatorDetails) internal _operatorDetails;
 
-    /// @notice Mapping: staker => operator whom the staker has delegated to
+    /**
+     * @notice Mapping: staker => operator whom the staker is currently delegated to.
+     * @dev Note that returning address(0) indicates that the staker is not actively delegated to any operator.
+     */
     mapping(address => address) public delegatedTo;
 
-    /// @notice Mapping: delegator => number of signed delegation nonce (used in delegateToBySignature)
-    mapping(address => uint256) public nonces;
+    /// @notice Mapping: staker => number of signed messages (used in `delegateToBySignature`) from the staker that this contract has already checked.
+    mapping(address => uint256) public stakerNonce;
```

Comment:
seems like it might be worth it to just set up a 
`mapping(address  => mapping(bytes32 => uint256)) nonces`
where `nonces[signer][typehash]` is their nonce for that struct they're signing?

Reply:
Interesting idea. Do you think this would be clearer for users/devs?
I feel like the existing design is "less clean" but the separation into different mappings/functions feels a little "safer"/nicer with respect to the dev UX? (slightly less footgunish, I think, with clearer differentation?)

Reply:
ah, i see. probably fine then. Another option would be do my mapping suggestion and have view functions for nonces. You can resolve.

Reply:
OK yeah, I think it's effectively the same, going to opt to leave as-is but I still appreciate the suggestion + discussion